helm/resource_release.go

```diff
@@ -140,7 +144,11 @@ func resourceRelease() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Default:     os.ExpandEnv("$HOME/.gnupg/pubring.gpg"),
-				Description: "Location of public keys used for verification.",
+				Description: "Location of public keys used for verification. Used only if `verify` is true",
+				// Suppress changes of this attribute as it is just a local path
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
> So we either set it to a default value that does not change, e.g `""` or we do not suppress changes when `verify` is true
@rporres Thanks for your reply! Yes, I was also thinking about these variants. But then I just din't find any reason why someone would need to store the local path to the keyring in the state (which is commonly shared). Do you know any cases when it is needed?

As I see, when several people work on the same terraform state, they should be able to use custom path to the keyring (via variables, for example) and that should not cause a resource update. Otherwise we run into #52.

FYI: Regardless of changes of this attribute, its value from the actual configuration is always passed to the Helm API when `verify` is true. So chart verification will always work with the keyring which is currently specified in Terraform config. Neither the old value nor the diff doesn't affect the resource behavior underneath.

Or maybe I miss something?

Reply:
I must say that I've never worked with chart verification myself, all I know is from reading the [helm provenance documentation](https://github.com/helm/helm/blob/master/docs/provenance.md). Reading that I understand where the default value comes from. I also think that if an organisation wants to work with chart verification, it should place the keyring (containing the public key associated with the signed chart) in the same place for every computer that runs terraform. That's why I think we should not ignore the changes in that value in case `verify` is true.

Does that make sense?

Reply:
Not really, I'm sorry :)

> That's why I think we should not ignore the changes in that value in case `verify` is true

As I said in the previous comment, the saved value of `keyring` attribute is not related to the resource behavior even when `verify` is true. Having this attribute changed doesn't necessarily mean that the keyring is changed (and vice versa). But luckily, the resource reads the actual content of the keyring file every time when it fetches the chart. So we can safely ignore changes of the local path in all cases.

But OK, if you believe we should keep tracking this attribute for the case when `verify` is true, then no problem - I'll add that quickly. But then we don't eliminate #52 completely.

Reply:
The `keyring` value is apparently used to verify the chart [locally](https://github.com/terraform-providers/terraform-provider-helm/blob/master/helm/resource_release.go#L721-L729) or [remotely](https://github.com/terraform-providers/terraform-provider-helm/blob/master/helm/resource_release.go#L770-L779). I must say I haven't tested myself that code and there are no tests for that, so I cannot say if it's working properly or if there's a bug there. I will take the action point of testing that and file an issue if it doesn't work.

In the meantime, we'd rather assume that the feature should not be broken so we should keep tracking this attribute for the case when `verify` is true. Thanks!

Reply:
OK. 👍
I'll pushed the new commit to the branch. It changes the `DiffSuppressFunc` logic.